src/vanilla/utils/atomWithStorage.ts

```diff
@@ -76,11 +76,16 @@ export function createJSONStorage<Value>(
   }
   if (
     typeof window !== 'undefined' &&
-    typeof window.addEventListener === 'function'
+    typeof window.addEventListener === 'function' &&
+    getStringStorage() instanceof window.Storage
```

Comment:
I don't think we should call `getString()` on creation. (It will fail on server.)

Reply:
It wouldn't run on the server because the condition will short-circuit on the checks for `window` before it reaches the call to `getStringStorage()`

Happy to remove this optimization though if it's a concern.

Reply:
Yes, please remove it. Some configs on sever add dummy `window`.

Reply:
Got it. :+1:  I reverted that part of my change and squashed the git history.